src/modules/composable-tasks/services/composable.tasks.transaction.ts

```diff
@@ -0,0 +1,175 @@
+import { EsdtTokenPayment } from '@multiversx/sdk-exchange';
+import { Injectable } from '@nestjs/common';
+import { MXProxyService } from 'src/services/multiversx-communication/mx.proxy.service';
+import {
+    ComposableTaskEnumType,
+    ComposableTaskType,
+} from '../models/composable.tasks.model';
+import { TransactionModel } from 'src/models/transaction.model';
+import {
+    BigUIntValue,
+    BytesType,
+    BytesValue,
+    EnumValue,
+    EnumVariantDefinition,
+    Field,
+    List,
+    ListType,
+    Struct,
+    TokenIdentifierValue,
+    TokenTransfer,
+    TypedValue,
+    U64Value,
+} from '@multiversx/sdk-core/out';
+import BigNumber from 'bignumber.js';
+import { gasConfig, mxConfig } from 'src/config';
+import { EgldOrEsdtTokenPayment } from 'src/models/esdtTokenPayment.model';
+import { decimalToHex } from 'src/utils/token.converters';
+import { WrapAbiService } from 'src/modules/wrapping/services/wrap.abi.service';
+
+export type ComposableTask = {
+    type: ComposableTaskType;
+    arguments: BytesValue[];
+};
+
+@Injectable()
+export class ComposableTasksTransactionService {
+    constructor(
+        private readonly mxPorxy: MXProxyService,
+        private readonly wrapAbi: WrapAbiService,
+    ) {}
+
```

Comment:
I would split this method into multiple smaller methods. for example getRawTasks and buildTransaction

Reply:
done